File: 2.annotations/src/moveread/annotations/images.py
from typing import Literal
from pydantic import ConfigDict
from moveread.boxes import Annotations as ImageAnnotations, Contours, Rectangle
from robust_extraction import Corners

Source = Literal['raw-scan', 'corrected-scan', 'camera', 'corrected-camera', 'robust-corrected'] 

class ImageMeta(ImageAnnotations):
  model_config = ConfigDict(extra='allow')
  source: Source | None = None
  perspective_corners: Corners | None = None

File: 2.annotations/ts/src/images.ts
export type ImageMeta = {
    box_contours?:        Array<Array<Array<number[]>>> | null;
    grid_coords?:         null | Rectangle;
    perspective_corners?: any[] | null;
    source?:              Source | null;
    [property: string]: any;
}


export type Source = "raw-scan" | "corrected-scan" | "camera" | "corrected-camera" | "robust-corrected";

  // hand-generated since quicktype treats `[number, number]` as `any[]`
  export type Rectangle = {
    size: [number, number]
    tl: [number, number]
  }
  